src/main/java/com/minecolonies/coremod/event/EventHandler.java

```diff
@@ -286,6 +293,9 @@ public void onPlayerInteract(@NotNull final PlayerInteractEvent.RightClickBlock
     {
         final EntityPlayer player = event.getEntityPlayer();
         final World world = event.getWorld();
+        BlockPos bedBlockPos = event.getPos();
+        final List<CitizenData> citizenList;
```

Comment:
I would get the event pos here yes, but I wouldn't put the world and citizen list here. Else we query the colony for each right click of a player, this is terrible performance wise

Reply:
The World object was already implemented in the code that way at the start, as it's used by several other events when the Player Right-Clicks. I'll move the Citizenlist object thought.

Reply:
sorry, I meant colony not world xD